### DISCUSSION TITLE
Porting a Mongo $expr query to Prisma

### DISCUSSION THREAD

**Discussion by viveleroi** [prisma/prisma#16935] [Q&A] [answered]
I am porting a MongoDB/mongoose app to Prisma and am struggling to understand how to concern some of our more complex queries. Maybe I'm just misreading the documentation or missing something but it seems like Prisma can't handle anything like the complexity that mongo can. For example, I am using mongo's `expr` to match the exact date/month/year of a date column:

```js
      $and: [
        { $expr: { $eq: [{ $year: '$date' }, date.getFullYear()] } },
        { $expr: { $eq: [{ $month: '$date' }, date.getMonth() + 1] } },
        { $expr: { $eq: [{ $dayOfMonth: '$date' }, date.getDate()] } }
      ]
```

I can't find equivalent to this in Prisma

**Comment by nurul3101** [comment, marked as answer]
Hello @viveleroi 👋 

You can use `$expr` by using [$runCommandRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#runcommandraw) method.
You can pass any parameters in `$runCommandRaw` function which are accepted by MongoDB's native `db.runCommand` Method: [MongoDB Docs](https://www.mongodb.com/docs/manual/reference/method/db.runCommand/)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
